### DISCUSSION TITLE
Sometimes, thumbnails fail to load

### DISCUSSION THREAD

**Discussion by freak12techno** [immich-app/immich#12644] [Q&A]
### The bug

I sometimes have thumbnails on some photos that are failing to load.
Once querying those, I am getting:
1) a web/mobile UI showing an error like this one:
<img width="208" alt="изображение" src="https://github.com/user-attachments/assets/c1ab4f84-4c86-406a-bf39-a43992e0571c">

2) a 500 error like `{"message":"Internal server error","statusCode":500,"correlationId":"ffsfyo55"}`
3) and this in logs:
```
[Nest] 17  - 09/13/2024, 1:43:11 AM   ERROR [Api:LoggerRepository~3kkdh7a7] Unable to send file: Error
Error: ENOENT: no such file or directory, access 'upload/thumbs/7eeb523e-601c-4b53-b469-fea0b8b86e22/c9/4c/c94ceda7-eca3-49ec-aa11-cc27b24b046d.webp'
    at async access (node:internal/fs/promises:606:10)
    at async sendFile (/usr/src/app/dist/utils/file.js:55:9)
    at async AssetMediaController.viewAsset (/usr/src/app/dist/controllers/asset-media.controller.js:57:9)
[Nest] 17  - 09/13/2024, 1:43:11 AM   ERROR [Api:GlobalExceptionFilter~3kkdh7a7] Unknown error: Error: ENOENT: no such file or directory, access 'upload/thumbs/7eeb523e-601c-4b53-b469-fea0b8b86e22/c9/4c/c94ceda7-eca3-49ec-aa11-cc27b24b046d.webp'
```

Pressing the "Refresh thumbnails" helps, but when pressing the "Missing" near the "Generate thumbnails" in the administration jobs page, it doesn't find these.

My concern here is that if it's a one-time thing (which it might be) it should be fixable with the "Generate thumbnails" => "Missing" job, but it isn't somehow.

Not sure what can go wrong here.

### The OS that Immich Server is running on

Unraid + docker-compose

### Version of Immich Server

v1.115.0

### Version of Immich Mobile App

v1.115.0

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
The default one
```

### Your .env content

```Shell
The default one
```

### Reproduction steps

See above.
Not sure how to reproduce it, as it just doesn't work sometimes.

### Relevant log output

```shell
See above.
```

**Comment by alextran1502** [comment]
When the thumbnail is failed to load with that message, it means that there was thumbnail generated and now cannot be read from the file system. Would it be possible that the mount to Immich on unraid gets unmounted somehow?

You can try regenerate thumbnail for a specific photos by selecting it and clicking on the context menu, then choose regenerate thumbnail

I suggest to figure out if the mount drive is somehow get unmounted and make sure that mechanism is solid to avoid future issues

**Reply by freak12techno** [reply]
This might happen, but wouldn't the whole Immich be inaccessible that way, given that it's mounted on the same volume (as far as I know)?

**Comment by radh21301** [comment]
Without the docker compose and env file, it would be really difficult to troubleshoot.

If you enter the console of immich-server, can you do a 
```
cd ./upload/thumbs/7eeb523e-601c-4b53-b469-fea0b8b86e22/c9/4c
```
and check the file it complains is indeed present?

**Reply by freak12techno** [reply]
Docker-compose: https://gist.github.com/freak12techno/df5a81f6e707e559bd83e98d112ea1be
Env: https://gist.github.com/freak12techno/f2ae726dc7868d070df38ee6437bac0b

On the error, I can't reproduce it on this specific file as I regenerated the thumbnail for it, here's the new error:
```
[Nest] 17  - 09/14/2024, 10:14:22 PM   ERROR [Api:LoggerRepository~8b7ir4wf] Unable to send file: Error
Error: ENOENT: no such file or directory, access 'upload/thumbs/7eeb523e-601c-4b53-b469-fea0b8b86e22/dd/2a/dd2ac65d-7ee9-4728-97e8-fbbf033d33b1.webp'
    at async access (node:internal/fs/promises:606:10)
    at async sendFile (/usr/src/app/dist/utils/file.js:55:9)
    at async AssetMediaController.viewAsset (/usr/src/app/dist/controllers/asset-media.controller.js:57:9)
```

and here's the files (not sure which one of the folders is correct, also I see it not from the container, but from the folder it's mounted to at the host machine):
```
root@shuttle:/mnt/user/main/immich/data# ls -la thumbs/7eeb523e-601c-4b53-b469-fea0b8b86e22/dd/2a/
total 1300
drwxr-xr-x 1 root root   4096 Aug 31 09:09 ./
drwxr-xr-x 1 root root   4096 Aug 30 11:56 ../
-rw-r--r-- 1 root root 201707 Aug 31 08:16 dd2a1812-29ad-42db-a9d2-0326de8bf746-preview.jpeg
-rw-r--r-- 1 root root 227826 Nov  2  2023 dd2a1812-29ad-42db-a9d2-0326de8bf746-thumbnail.jpeg
-rw-r--r-- 1 root root   5928 Aug 31 09:04 dd2a1812-29ad-42db-a9d2-0326de8bf746-thumbnail.webp
-rw-r--r-- 1 root root 374626 Nov  2  2023 dd2ac65d-7ee9-4728-97e8-fbbf033d33b1-thumbnail.jpeg
-rw-r--r-- 1 root root 238136 Aug 31 08:20 dd2ad983-8f28-4c9e-ab62-d021bf86db28-preview.jpeg
-rw-r--r-- 1 root root 253533 Nov  2  2023 dd2ad983-8f28-4c9e-ab62-d021bf86db28-thumbnail.jpeg
-rw-r--r-- 1 root root   6532 Aug 31 09:09 dd2ad983-8f28-4c9e-ab62-d021bf86db28-thumbnail.webp
root@shuttle:/mnt/user/main/immich/data# ls -la upload/7eeb523e-601c-4b53-b469-fea0b8b86e22/dd/2f
/bin/ls: cannot access 'upload/7eeb523e-601c-4b53-b469-fea0b8b86e22/dd/2f': No such file or directory
```

**Reply by radh21301** [reply]
To clarify - are you saying you cant find that file when you look for it inside the container? But can find it on the hist machine?

That would be odd because you can see the thumbnail of the file you just regenerated.

**Reply by freak12techno** [reply]
I mean, I did not do it from within the container, what I am showing is from the host machine from the folder which is mounted.

**Reply by radh21301** [reply]
I dont have an exact solution. But did you by any chance switch from to jpeg to webp for thumbnails Or vice versa?
I had similar issues I think. And it got sorted when I deleted the content of the thumbnails folder and ran the thumbnail job on all assets. But that might not be a feasible option if you have too many assets.

**Reply by freak12techno** [reply]
No, I for sure did not, at least manually.

**Reply by radh21301** [reply]
That is odd to me. Because a couple of messages earlier, in the   Terminal output, I see both .webp and .jpeg thumbnails with the same name

dd2ad983-8f28-4c9e-ab62-d021bf86db28-thumbnail.jpeg

dd2ad983-8f28-4c9e-ab62-d021bf86db28-thumbnail.webp

Not sure what is going on.

**Reply by freak12techno** [reply]
@radh21301 actually no, the .webp is missing, and that's what I think causes it. Not sure why though.
